### ISSUE TITLE
User can accept txs with insufficient funds and sometimes no callback

### ISSUE THREAD

**Issue by flyswatter** [MetaMask/metamask-plugin#660]
Apparently we sometimes do not call back an error when a user tries to accept a tx they do not have sufficient funds to complete.

Could be solved by #519

**Comment by frankiebee** [comment]
Half of this is fixed, but the other half would be solved by giving access to the account balanced while the account is locked and throwing an error before showing confirmation screen.

**Comment by frankiebee** [comment]
We want to throw an error if the user does not have enough funds..? Well at the moment you cant ```approve``` a transaction at all if the transaction is more then the balance so  is the prposed flow be that it doesn't even show the confirmation screen and just shows an error screen?

**Comment by flyswatter** [comment]
This feature has been requested, I think it makes sense, but we don’t have access to balances when locked until after the multivault refactor.

**Comment by frankiebee** [comment]
This is blocked by issue #328

**Comment by flyswatter** [comment]
We *could* start doing this on the `dev` branch now.  Probably want to wait for it to be merged into master first.

**Comment by flyswatter** [comment]
Closing this because we're leaning towards a log-in-per-site, as suggested in #799.
